src/Optical/OpticalSystem.jl

```diff
@@ -147,79 +158,7 @@ function trace(
         return nothing
     end
 
-    result = trace(system.assembly, r, temperature(system), pressure(system), trackrays = trackrays, test = test)
```

Comment:
Not understanding how you can delete this. It's not the same as the trace in LensAssembly. It does the memory managment for the interval pools, for example.